controllers/platformnetwork_controller.go

```diff
@@ -1325,3 +1334,56 @@ func (r *PlatformNetworkReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.PlatformNetwork{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *PlatformNetworkReconciler) checkRestoreInProgress(instance *starlingxv1.PlatformNetwork) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status and clear annoatation
+func (r *PlatformNetworkReconciler) RestorePlatformNetworkStatus(instance *starlingxv1.PlatformNetwork) error {
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	config, ok := annotation[cloudManager.RestoreInProgress]
+	if ok {
+		restoreStatus := &cloudManager.RestoreStatus{}
+		err := json.Unmarshal([]byte(config), &restoreStatus)
+		if err == nil {
+			if restoreStatus.InSync != nil {
+				instance.Status.InSync = *restoreStatus.InSync
+			}
+			if restoreStatus.Reconciled != nil {
+				instance.Status.Reconciled = *restoreStatus.Reconciled
+			}
+			if restoreStatus.DeploymentScope != nil {
+				if strings.ToLower(*restoreStatus.DeploymentScope) == "bootstrap" ||
+					strings.ToLower(*restoreStatus.DeploymentScope) == "principal" {
+					instance.Status.DeploymentScope = *restoreStatus.DeploymentScope
+				} else {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Invalid deploymentScope value '%s', scope will be set to boostrap", *restoreStatus.DeploymentScope)
+					instance.Status.DeploymentScope = "bootstrap"
+				}
+			}
+			err = r.Client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				return common.NewResourceStatusDependency(fmt.Sprintf("Failed to update platform network status while restoring '%s' platform network resource.", instance.Name))
+			} else {
```

Comment:
else is not needed since the if statement already includes a return.

Reply:
If there is an error, it returns an error. If there is no error, we are just logging the successful status update with the values